pkgs/top-level/default.nix

```diff
@@ -58,8 +69,14 @@ in let
   nixpkgsFun = newArgs: import ./. (args // newArgs);
 
   # Partially apply some args for building phase pkgs sets
-  pkgs = import ./stage.nix ({
-    inherit lib nixpkgsFun config platform;
+  allPackages = args: import ./stage.nix ({
+    inherit lib nixpkgsFun config;
   } // args);
 
+  inherit (import ../stdenv {
+    inherit lib allPackages system platform crossSystem config;
+  }) stdenv;
+
+  pkgs = allPackages { inherit system stdenv config crossSystem platform; };
```

Comment:
Note:
This commit changes the dependencies of `stdenv`, and clean-up the stdenv story by removing the `defaultStdenv` attribute as well as the `bootStdenv` argument.

Before, stdenv was provided by all-packages, which was iterating multiple time over the `top-level/default.nix` expression.  This commit changes this relation such that the `stdenv` used for building all the *final* packages depends on the `allPackages` function, which does a fix-point over the set of packages needed for the bootstrap.  And the *final* set of packages depends on the `allPackages` function and on the `stdenv` result.